chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1058 @@
+
+
+def parse_response(r):
+    ck = sum(r[:-1]) % 256
+    if ck != r[-1]:
+        err = f'Data check sum failed calc {ck} got {r[-1]}'
+        log.error(err)
+    return r[5:-1]
+
+
+def get_ident(serial):
+    try:
+        bs = send_cmd(serial, b'\x02', 16)
+        model, mode, ver = struct.unpack('>x7sB4s3x', bs)
+        log.info(f'Model {model} Ver {ver} Band Mode {mode}')
+        if ver in SUPPORTED_MODEL_VERSIONS[model]:
+            log.info(f'Model {model} Ver: {ver} is supported')
+        else:
+            err = f'ERROR: Model: {model} Ver: {ver} is not supported!'
+            log.error(err)
+            raise errors.RadioError(err)
+        return bs
+    except Exception as e:
+        raise errors.RadioError(e)
+
+
+def get_bands(mode):
+    # Radio skips 0xA-0XF, 0x10 == 10 :/
+    try:
+        idx = int(hex(mode)[2:]) - 1
+        valid_ranges = BAND_LIMITS[idx]
+    except IndexError:
+        raise errors.RadioError("Unknown Radio Mode - Can't set Freq. Bands!")
+    return valid_ranges
+
+
+def do_download(radio):
+    try:
+        data = b''
+        serial = radio.pipe
+        serial.timeout = SERIAL_TIMEOUT
+        status = chirp_common.Status()
+        status.msg = "Connecting to Radio..."
+        radio.status_fn(status)
+        r = start_programming(serial)
+        if not r:
+            raise errors.RadioError('Failed to enter programming mode')
+        ident_bytes = get_ident(serial)
+        data = ident_bytes
+        status.max = sum((t[1] * t[2]) for t in MEMORY_REGIONS if t[3] != 'S')
+        for reg in MEMORY_REGIONS:
+            base, blocks, blk_sz, mod, struc, label = reg
+            if mod != 'S':
+                status.msg = f'Downloading: {label}...'
+                addrs = [(base + i) for i in range(0, blocks)]
+                for addr in addrs:
+                    cmd = struct.pack('>BL', 0x52, addr)
+                    r = send_cmd(serial, cmd, blk_sz + 6)
+                    if r[0] != 0x57:
+                        err = (f'Received bad ACK from Radio '
+                               f'expect 0x57 got {hex(r[0])}')
+                        log.error(err)
+                        raise errors.RadioError(err)
+                    data += parse_response(r)
+                    status.cur += len(r[5:-1])
+                    radio.status_fn(status)
+    except Exception as e:
+        raise errors.RadioError(f'Exception During Download: {e}')
+    finally:
+        end_programming(serial)
+    return memmap.MemoryMapBytes(data)
+
+
+def write_data(serial, addr, data):
+    pfx = struct.pack('>BL', 0x57, addr)
+    data = pfx + data
+    chk_int = sum(data) % 256
+    chk_byte = struct.pack('>B', chk_int)
+    cmd = data + chk_byte
+    r = send_cmd(serial, cmd, 1)
+    if r != b'\x06':
+        err = (f'Received bad ACK from Radio '
+               f'expect 0x06 got {hex(r[0])}')
+        log.error(err)
+        raise errors.RadioError(err)
+    return
```

Comment:
This is a nit, but it's conventional in python not to put a naked return like this when you don't need it :)